src/hera/shared/_global_config.py

```diff
@@ -24,7 +24,7 @@ class _GlobalConfig:
     host: Optional[str] = None
     verify_ssl: bool = True
     api_version: str = "argoproj.io/v1alpha1"
-    namespace: Optional[str] = None
+    namespace: Optional[str] = "default"
```

Comment:
This is worthy of review

Reply:
I think we had this as `None` because from [the Argo spec](https://argoproj.github.io/argo-workflows/fields/#objectmeta), a missing namespace *is* "default":
>Namespace defines the space within which each name must be unique. An empty namespace is equivalent to the "default" namespace, but "default" is the canonical representation. Not all objects are required to be scoped to a namespace - the value of this field for those objects will be empty.Must be a DNS_LABEL. Cannot be updated. More info: http://kubernetes.io/docs/user-guide/namespaces

So I think it's fine either way but I would argue it's an extra bit of logic to maintain!

Reply:
I did not know that! Thanks for pointing out. The only reason I added it, tbh, is that upon submitting workflows via `w.create()` I got errors on local that `namespace` is `None`. How about we keep it for that specific use case but we also link this documentation as justification for adding `default`?

Reply:
I would prefer if we did not default to the `default` namespace if possible. 

It makes it difficult to provide re-usable/copy-pastable examples or code.

Related - https://release.com/blog/10-kubernetes-namespace-best-practices-to-start-following#:~:text=Don't%20Use%20the%20Default,or%20implement%20RBAC%20and%20NetworkPolicies.
<img width="960" alt="image" src="https://user-images.githubusercontent.com/16130816/227807491-872fab86-dad7-4226-b19a-a76c98a07e2f.png">

If no namespace is provided it would default to the `default` namespace as @elliotgunton mentioned.